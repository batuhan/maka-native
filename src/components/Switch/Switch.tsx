import React, { ReactNode } from "react";
import { Switch } from "react-native";
import styled from "styled-components";

interface Props {
  style?: {};
  disabled: boolean;
  onValueChange: () => void;
}

export default function({ disabled = false, style, onValueChange }: Props) {
  return (
    <Switch style={style} onValueChange={onValueChange} disabled={disabled} />
  );
}
