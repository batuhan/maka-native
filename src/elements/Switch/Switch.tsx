import React, { ReactNode } from "react";
import { Switch } from "react-native";
import styled from "styled-components";

interface Props {
  style?: {};
  disabled: boolean;
  onValueChange: (value: boolean) => void;
  value: boolean;
}

export default function ({ disabled = false, style, onValueChange, value }: Props) {
  return (
    <Switch style={style} onValueChange={onValueChange} disabled={disabled} value={value} />
  );
}
