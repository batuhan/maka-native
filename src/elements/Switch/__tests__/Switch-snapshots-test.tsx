import React from "react";
import renderer from "react-test-renderer";

import Switch from "../Switch";

describe("Switch", () => {
  it("renders correctly", () => {
    const disabled = false;
    const onValueChange = jest.fn();
    const instance = renderer.create(
      <Switch disabled={disabled} onValueChange={onValueChange} value={false} />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
