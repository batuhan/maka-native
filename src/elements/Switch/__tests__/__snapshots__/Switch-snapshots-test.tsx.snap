// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
<RCTSwitch
  accessibilityRole="button"
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
/>
`;
